$NetBSD: patch-ag,v 1.1 2006/01/18 12:44:03 jmmv Exp $

Removed the #define directive from the inside of a macro expansion.
Needed for gcc-2.95 and possibly others.

--- sys/glsink/glimagesink.c.orig	2005-06-14 18:38:03.000000000 +0200
+++ sys/glsink/glimagesink.c
@@ -46,6 +46,12 @@ GST_ELEMENT_DETAILS ("Video sink",
     "An OpenGL 1.2 based videosink",
     "Gernot Ziegler <gz@lysator.liu.se>, Julien Moutte <julien@moutte.net>");
 
+#ifdef GL_YCBCR_MESA
+#  define YUV_FOO ";" GST_VIDEO_CAPS_YUV ("{ UYVY, YUY2 }")
+#else
+#  define YUV_FOO
+#endif
+
 /* Default template - initiated with class struct to allow gst-register to work
    without X running */
 static GstStaticPadTemplate gst_glimagesink_sink_template_factory =
@@ -53,12 +59,12 @@ static GstStaticPadTemplate gst_glimages
     GST_PAD_SINK,
     GST_PAD_ALWAYS,
     GST_STATIC_CAPS (GST_VIDEO_CAPS_RGBx ";" GST_VIDEO_CAPS_BGRx
-#ifdef GL_YCBCR_MESA
-        ";" GST_VIDEO_CAPS_YUV ("{ UYVY, YUY2 }")
-#endif
+    YUV_FOO
     )
     );
 
+#undef YUV_FOO
+
 #if 0
 enum
 {
